contracts/access/manager/AccessManager.sol

```diff
@@ -0,0 +1,339 @@
+        uint8 group,
+        bool allowed
+    ) public virtual onlyDefaultAdmin {
+        require(_contractMode[target] == RestrictedMode.Custom, "AccessManager: target in special mode");
+        for (uint256 i = 0; i < selectors.length; i++) {
+            bytes4 selector = selectors[i];
+            _allowedGroups[target][selector] = _withUpdatedGroup(_allowedGroups[target][selector], group, allowed);
+            emit GroupAllowed(target, selector, group, allowed);
+        }
+    }
+
+    /**
+     * @dev Returns the mode of the target contract, which may be custom (`0`), closed (`1`), or open (`2`).
+     */
+    function getContractMode(address target) public view virtual returns (RestrictedMode) {
+        return _contractMode[target];
+    }
+
+    /**
+     * @dev Sets the target contract to be in custom restricted mode. All restricted functions in the target contract
+     * will follow the group-based restrictions defined by the AccessManager. The caller must be the default admin.
+     */
+    function setContractModeCustom(address target) public virtual onlyDefaultAdmin {
+        _setContractMode(target, RestrictedMode.Custom);
+    }
+
+    /**
+     * @dev Sets the target contract to be in "open" mode. All restricted functions in the target contract will become
+     * callable by anyone. The caller must be the default admin.
+     */
+    function setContractModeOpen(address target) public virtual onlyDefaultAdmin {
+        _setContractMode(target, RestrictedMode.Open);
+    }
+
+    /**
+     * @dev Sets the target contract to be in "closed" mode. All restricted functions in the target contract will be
+     * closed down and disallowed to all. The caller must be the default admin.
+     */
+    function setContractModeClosed(address target) public virtual onlyDefaultAdmin {
+        _setContractMode(target, RestrictedMode.Closed);
+    }
+
+    /**
+     * @dev Transfers a target contract onto a new authority. The caller must be the default admin.
+     */
+    function transferContractAuthority(address target, address newAuthority) public virtual onlyDefaultAdmin {
+        AccessManaged(target).setAuthority(IAuthority(newAuthority));
+    }
+
+    /**
+     * @dev Creates a new group.
+     *
+     * Emits {GroupUpdated}.
+     */
+    function _createGroup(uint8 group, string memory name) internal virtual {
+        require(!hasGroup(group), "AccessManager: existing group");
+        _createdGroups = _withUpdatedGroup(_createdGroups, group, true);
+        emit GroupUpdated(group, name);
+    }
+
+    /**
+     * @dev Augmented version of {AccessControl-_grantRole} that keeps track of user group bitmaps.
+     */
+    function _grantRole(bytes32 role, address user) internal virtual override {
+        super._grantRole(role, user);
+        (bool isGroup, uint8 group) = _decodeGroupRole(role);
+        if (isGroup) {
+            require(hasGroup(group), "AccessManager: unknown group");
```

Comment:
I think this `require` should be a noop instead. The probability of `isGroup` being hit by accident is `2 ** (8 * 6)`, which I consider non-negligible.

Specially because some people use the `abi.encode(bytes("..."))`, which may have `group:` as prefix in some setups. 

Look at [this one](https://github.com/OpenZeppelin/accesscontrol-explorer/blob/master/src/lib/role-dict.json#L5270) that starts with `gr` for `grantor`.

Reply:
Nevermind, I realized removing this restriction will make easy to impersonate a group by crafting a role that starts with the 6 bytes of `group:`. Think about cases like the following may not be easy to spot:

```
bytes32 SOME_ROLE = abi.encodePacked(bytes("group:..."));
```

And we've seen such cases https://github.com/OpenZeppelin/accesscontrol-explorer/blob/master/src/lib/role-dict.json#L5270